modules/sdk-core/src/bitgo/lightning/lightning.ts

```diff
@@ -114,4 +118,32 @@ export class Lightning implements ILightning {
       throw new Error(`error(s) parsing response body: ${errors}`);
     });
   }
+
+  public async decodeLnurlPay(lnurl: string): Promise<LnurlPayResponse & { domain: string }> {
```

Comment:
I like the way you've done it with `LnurlPayResponse & { domain: string }` per our earlier conversation about `domain`. I might go one step further and give this type a name, maybe `type DecodedLnurlPayRequest = LnurlPayResponse & { domain: string }` which is used here. I think that's a little bit clearer.

Edit: Even better, I think we should differentiate the units on the LNURL response per the spec (millisats) from the units we're working with (sats). So ideally the decoded type should look like this:

```typescript
type DecodedLnurlPayRequest = Omit<LnurlPayResponse, 'maxSendable' | 'minSendable'> & { 
  domain: string,
  /** The maximum amount in satoshis we can pay for this LNRUL request /**
  maxSendableSats: number,
  /** The minimum amount in satoshis we can pay for this LNRUL request /**
  minSendableSats: number,
}
```